src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ParameterNamesShouldMatchBaseDeclaration.cs

```diff
@@ -136,11 +136,14 @@ private static ImmutableArray<IMethodSymbol> GetOriginalDefinitions(IMethodSymbo
             }
 
             var typeSymbol = methodSymbol.ContainingType;
+            var methodSymbolName = methodSymbol.Name;
 
             originalDefinitionsBuilder.AddRange(typeSymbol.AllInterfaces
-                .SelectMany(m => m.GetMembers(methodSymbol.Name))
-                .Where(m => typeSymbol.FindImplementationForInterfaceMember(m) != null)
-                .Cast<IMethodSymbol>());
+                .SelectMany(m => m.GetMembers(methodSymbolName))
+                .OfType<IMethodSymbol>()
+                .Where(m => methodSymbol.Parameters.Length == m.Parameters.Length
```

Comment:
Ideally, we would call `Microsoft.CodeAnalysis.CSharp.Symbols.MemberSignatureComparer.CSharpImplicitImplementationComparer.Equals` to verify the method being selected is compatible with `methodSymbol`. However, this functionality is internal to the `Microsoft.CodeAnalysis.CSharp` assembly and we cannot access it from here. This is an approximation: by checking the number of parameters and method arity. This fixes crashes in the rule.